app/services/trust_pilot_parser.rb

```diff
@@ -0,0 +1,67 @@
+class TrustPilotParser
+  attr_reader :reviews
+
+  def initialize(data)
+    @reviews = data['reviews'] || []
+  end
+
+  def parse_reviews
+    reviews.map do |review|
+      ReviewParser.new(review).parse
+    end
+  end
+
+  class ReviewParser
+    ATTRIBUTES = %w[response_id score comment title language created_at updated_at company_reply
+                    is_verified number_of_likes status report_data compliance_labels
+                    consumer_name consumer_location email user_reference_id].freeze
+    attr_reader :data
+
+    def initialize(data)
+      @data = OpenStruct.new(data)
+    end
+
+    def parse
+      ATTRIBUTES.inject({}) { |acc, key| acc.merge(key => extract_data(key)) }
+    end
+
+    private
+
+    def extract_data(key)
+      attr_name = key.camelize(:lower)
+      if data.respond_to?(attr_name)
+        data.send(attr_name)
+      else
+        send(key)
+      end
+    end
+
+    def response_id
```

Comment:
we can use `delegate`, like:
`delegate :response_id , :id, :score, :comment,  :email, to: :data`

Reply:
I tried with delegate, but data hasn't that keys, besides they are all camelcase, so i needed change to snakecase

Reply:
you can do this in Parser class, in top i have comment. then here we can just delegate. check my comment about CustomOpenstruct